### PR TITLE
Add hover card to cliff rectangle

### LINKED ISSUE #701: Add hovercard for cliff rectangles
For example:
>This household's net income falls for earnings between $19,400 and $23,100 {in the baseline/reform/baseline and reform scenario(s)}.
image

### PR DESCRIPTION
**Pull request by MaxGhenis**
Fixes #701 

Here's how it looks:
image

I'd still like layer it under the lines, so users can still see the net income values.

### CODE REVIEW

**Comment by nikhilwoodruff** [comment]
Looks amazing, thanks

**Comment by MaxGhenis** [comment]
@nikhilwoodruff do you know how to instantiate the figure from the `go.Scatter` so that the shading is "below" the lines? I suspect this is pretty easy but I'm not too familiar with `graph_objects`.

**Comment by nikhilwoodruff** [comment]
Ah, sorry not offhand

**Comment by MaxGhenis** [comment]
I got the line above the rectangle, but now it lost the hovercard.
image

**Comment by MaxGhenis** [comment]
Success!
image
image
image

File: policyengine/impact/household/charts.py [hunk 1]
```diff
@@ -461,8 +478,11 @@ def mtr_chart(
         line_shape="hv",
     )
     # Shade baseline and reformed net income cliffs.
-    shade_cliffs(baseline, config, fig, charts.GRAY)
-    shade_cliffs(reformed, config, fig, charts.BLUE)
+    ymax = df[y_fig].max() * 1.05  # Add a buffer.
```
Comment (nikhilwoodruff) [inline policyengine/impact/household/charts.py:481]: Sounds good, any particular reason for 5% or is just arbitrary?

**Review by nikhilwoodruff — APPROVED** [review]

File: policyengine/impact/household/charts.py (hunk 1 above)
Comment (MaxGhenis) [reply policyengine/impact/household/charts.py:481]: Arbitrary, tends to align with what plotly makes the y axis with respect to the max value IME

**Pull request merged**